### DISCUSSION TITLE
gather device group facts panos_device_group

### DISCUSSION THREAD

**Discussion by Christian-Kolrep** [PaloAltoNetworks/pan-os-ansible#393] [Q&A]
Hi, i want to gather facts of configured device groups in panorama. But i'm missing some "gathered" examples in the docs. My task looks like this:

> - name: get device  group facts
   paloaltonetworks.panos.panos_device_group:
    provider: "{{ paloalto_provider }}"
    state: "gathered"
    gathered_filter: '*'
  connection: local
  register: dg_facts

But the content looks strange. I'm trying to get the name of all device groups and the associated member devices/vsys.
I am grateful for any help

**Comment by jamesholland-uk** [comment]
The output today will be the Device Group names (and the tags configured inside that Device Group). One could probably use `panos_op` with `cmd: "show devicegroups name {{ item.name }}"` to further this and get the devices/VSYS per Device Group, with some nested looping. What's the usage for this, there may be a better way...
